### PR TITLE
fix(wallet): Update amount to send decimal handling

### PR DESCRIPTION
**Pull request by Cuteivist**
Fixes https://github.com/status-im/status-desktop/issues/14163

### What does the PR do

* Removed hard-coded `en_US` locale. Other readonly fields use system locale, so it creates weirdness where different decimal characters are used in inputs
* Added sub-component that handles amount characters for StatusInput
* Updated all the fields that used hard-coded locale

### Affected areas

* Wallet / send modal
* Community airdrop

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/11396062/6198949c-b44b-4c25-a019-d446f877a55b

### CODE REVIEW

File: ui/app/AppLayouts/Communities/controls/TokenPanel.qml [hunk 1]
```diff
@@ -138,8 +138,17 @@ ColumnLayout {
             if(!addOrUpdateButton.enabled)
                 return
 
-            if(event.key === Qt.Key_Enter || event.key === Qt.Key_Return)
+            // additionally accept dot (.) and convert it to the correct decimal point char
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/controls/TokenPanel.qml:141]: Can't use the new `StatusAmountInput` here as well? :thinking: That would reduce the code duplication

File: ui/imports/shared/popups/send/SendModal.qml [hunk 2]
```diff
@@ -504,8 +504,7 @@ StatusDialog {
                 totalTokenFeesInFiat = gasTimeEstimate.totalTokenFees * d.selectedHolding.marketDetails.currencyPrice.amount
             d.totalFeesInFiat = d.currencyStore.getFiatValue(gasTimeEstimate.totalFeesInEth, Constants.ethToken) + totalTokenFeesInFiat
 
-            if (!!d.selectedHolding.type && (d.selectedHolding.type === Constants.TokenType.ERC20
-                                             || d.selectedHolding.type === Constants.TokenType.ETH)) {
+            if (d.selectedHolding.type === Constants.TokenType.ERC20 || d.selectedHolding.type === Constants.TokenType.Native) {
```
Comment (caybro) [inline ui/imports/shared/popups/send/SendModal.qml:507]: What's this change, seems unrelated?

File: ui/imports/shared/popups/send/views/AmountToSend.qml [hunk 3]
```diff
@@ -131,8 +131,10 @@ ColumnLayout {
             id: topAmountToSendInput
             Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
             Layout.maximumWidth: 250
-            Layout.preferredWidth: !!text ? input.edit.paintedWidth
+            // Additional padding for cursor
+            Layout.preferredWidth: !!text ? input.edit.paintedWidth + 2
                                           : textMetrics.advanceWidth
+            locale: LocaleUtils.userInputLocale
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/AmountToSend.qml:137]: You can remove this, it's the default

File: ui/imports/shared/popups/send/views/AmountToSend.qml [hunk 4]
```diff
@@ -131,8 +131,10 @@ ColumnLayout {
             id: topAmountToSendInput
             Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
             Layout.maximumWidth: 250
-            Layout.preferredWidth: !!text ? input.edit.paintedWidth
+            // Additional padding for cursor
+            Layout.preferredWidth: !!text ? input.edit.paintedWidth + 2
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/AmountToSend.qml:135]: Put the padding in `AmountInputWithCursor`?

**Review by caybro — COMMENTED** [review]
Overall great work, thanks! Agreed with @Cuteivist to also create a storybook page for the new component

File: ui/StatusQ/src/StatusQ/Controls/StatusAmountInput.qml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    ]
+
+    onKeyPressed: (event) => {
+                      // additionally accept dot (.) and convert it to the correct decimal point char
+                      if (event.key === Qt.Key_Period || event.key === Qt.Key_Comma) {
+                          // Only one decimal point is allowed
+                          if(root.input.text.indexOf(root.locale.decimalPoint) === -1)
+                            root.input.insert(root.input.cursorPosition, root.locale.decimalPoint)
+                          event.accepted = true
+                      } else if ((event.key > Qt.Key_9 && event.key <= Qt.Key_BraceRight) || event.key === Qt.Key_Space) {
```
Comment (saledjenic) [inline ui/StatusQ/src/StatusQ/Controls/StatusAmountInput.qml:30]: Why `Qt.Key_Space` here?

**Review by saledjenic — APPROVED** [review]

**Review by caybro — APPROVED** [review]
Great work, thanks for the SB page & tests! :beers:
